x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1109,10 +1109,7 @@ public AsyncSender interceptSender(AsyncSender sender) {
 
     @Override
     public UnaryOperator<RestHandler> getRestHandlerWrapper(ThreadContext threadContext) {
-        if (enabled == false) {
-            return null;
-        }
-        final boolean ssl = HTTP_SSL_ENABLED.get(settings);
+        final boolean ssl = enabled && HTTP_SSL_ENABLED.get(settings);
```

Comment:
This is the main part of the PR - it's the thing that actually achieves the result we're after, but I don't see anywhere where we test that this PR delivers the behaviour that we've decided on.

That is, we need a `testSecurityHandlerIsAlwaysInstalled`